src/views/open-api-docs-view/components/property-details/index.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import classNames from 'classnames'
+import { MdxInlineCode } from 'components/dev-dot-content/mdx-components'
+import Badge from 'components/badge'
+import { IconCornerDownRight16 } from '@hashicorp/flight-icons/svg-react/corner-down-right-16'
+import s from './property-details.module.css'
+
+/**
+ * Types are shared by the "base" and "deeper" variations of
+ * the property details component.
+ */
+export type PropertyDetailsProps = {
+	name: string
+	type: string
+	isRequired?: boolean
+	description?: string // Plain text or HTML
+	nestedProperties?: PropertyDetailsProps[]
+	depth?: number
+}
+
+/**
+ * Render a top-level property, with optional nested properties.
+ *
+ * Future: will need to support "beta" badge for HCP Packer docs.
+ */
+export function PropertyDetails({
+	name,
+	type,
+	isRequired,
+	description,
+	nestedProperties,
+	depth = 0,
+}: PropertyDetailsProps) {
+	return (
+		<div className={s.baseRoot}>
+			<div className={s.baseMetaAndDescription}>
+				<div className={s.baseMeta}>
+					<MdxInlineCode>{name}</MdxInlineCode>
+					<span className={s.baseType}>{type}</span>
+					{isRequired ? <Badge text="Required" color="highlight" /> : null}
+				</div>
+				{description ? (
+					<div
+						className={s.baseDescription}
+						dangerouslySetInnerHTML={{ __html: description }}
+					/>
+				) : null}
+			</div>
```

Comment:
<img width="555" alt="Screenshot 2023-08-02 at 11 33 58 AM" src="https://github.com/hashicorp/dev-portal/assets/36613477/2204b405-2c11-4113-8571-5c1d552a0f67">
<img width="441" alt="Screenshot 2023-08-02 at 11 35 01 AM" src="https://github.com/hashicorp/dev-portal/assets/36613477/c80e8895-21fd-47cf-863a-4484687e264e">

Makes sense if we don't want to do this style now, but just noting the discrepancy.

Reply:
Ah yeah, I should have made an explicit note of this! For now we're going with static display rather than interactive form 👍

Reply:
Sorry @kendallstrautman I should have made that more clear. Future ideas ⭐